azavea/sentinel-2-stac-utils
# Enable creating catalog for Sentinel 2 L2A

## Overview

This PR, building on top of commands for creating catalog of Sentinel 2 L1C, enables creating catalog of Sentinel 2 L2A. It turns out the data models are re-usable, since `productInfo` and `tileInfo` of those two products share the same json schema, yay :tada: -- so you will see some renaming. The bigger distinction is how bands are stored on S3  -- L2A has bands separated in different folders differentiated by resolution. So, this PR passes the distinction down to the crawler and include the highest resolution available for each band to be included as the image assets. Another distinction is where the `TCI` asset is -- L2A also has it under resolution-specific folders, this PR uses the `R10m` one, which is the highest resolution available afaik.

### Checklist

- ~[] New tests have been added or existing tests have been modified~ (???)

### Testing Instructions

- Instructions here: https://github.com/azavea/sentinel-2-stac-utils/pull/7#issue-453595216 are v nice, following them should test this PR too! Just replace the commands with the following to test writing to S3 and to local, respectively:

`AWS_PROFILE=raster-foundry AWS_REGION=us-east-1 bloop run s2stac -- create --collection L2A --inventory-path aaron_is_the_best --output-catalog-root <s3 location you can write>`

`AWS_PROFILE=raster-foundry AWS_REGION=us-east-1 bloop run s2stac -- create --collection L2A --inventory-path aaron_is_the_best --output-catalog-root <a local directory>`

Closes https://github.com/azavea/sentinel-2-stac-utils/issues/3
